docs/coding-guidelines.md

```diff
@@ -239,7 +239,36 @@ For example the following are correct:
     }
     ```
 
-Exceptions are allowed when multiple type names coming from different namespaces are available.
+    Exceptions are allowed when multiple type names coming from different namespaces are available.
+
+1. Do not use `as` to cast types
+
+   These are correct:
+
+   ```cs
+   ListItem item = (ListItem)selectedItem; // InvalidCastException if wrong type
+   string name = item.Name;
+
+   if (obj is Type1 t1)
```

Comment:
What is this `obj` here? Above 2 lines are related to this one? Looks like 2 different use cases. If it's 2 different cases could you please separate them?

Reply:
I'm not sure why GitHub's not marking the code snippet as "outdated", but I put the two samples in different code blocks, even though that pattern isn't used elsewhere in the document.